Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM ubuntu:latest
+
+# Install environment dependencies
+RUN apt update && apt install -y git wget unzip python3 python3-pip python3-dev python-is-python3 default-jdk nodejs npm
+RUN python3 -m pip install -U pip
+
+# Setup build environment
+ENV BAZEL_VERSION='4.0.0'
+ENV BAZEL_SHA256SUM='7bee349a626281fc8b8d04a7a0b0358492712377400ab12533aeb39c2eb2b901'
+ENV BUILDTOOLS_VERSION='3.0.0'
+ENV BUILDIFIER_SHA256SUM='e92a6793c7134c5431c58fbc34700664f101e5c9b1c1fcd93b97978e8b7f88db'
+ENV BUILDOZER_SHA256SUM='3d58a0b6972e4535718cdd6c12778170ea7382de7c75bc3728f5719437ffb84d'
+ENV TENSORFLOW_VERSION='tf-nightly'
+
+# Get the code
+RUN git clone https://github.com/tensorflow/tensorboard /tensorboard
+WORKDIR /tensorboard
```

Comment:
Each [RUN](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers) directive creates a new image layer. Consider consolidating all of the build steps into a single `RUN` command.

At minimum, I'd suggest doing the more costly build steps that don't change often (i.e. download and installbazel,ibazel) before the things that change more frequently (pip dependencies, getting the most up-to-date code) to leverage the [build cache](https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#leverage-build-cache) more effectively.

Reply:
I'm not sure why consolidating the build steps would utilize the build cache more effectively, wouldn't more granular steps be better for caching?

Reply:
Apologies, those were separate suggestions, but that was definitely not clear.

After some offline discussion, it definitely doesn't make sense to have a monolithic `RUN` directive.

More granular steps would be good for caching as long as they're ordered properly (i.e. less frequently-updated steps come earlier).